vars/generateAndExecutePipeline.groovy

```diff
@@ -119,13 +134,56 @@ def _getApplicationDefinitionFilePath() {
     return 'applicationDefinition.yaml'
 }
 
+def _validateReleaseBranch(applicationDefinition, scmMetadata) {
+
+    if (isReleaseBranch()==true) {
+
+        def majorVersion = applicationDefinition.Pipeline.MajorVersion
+        def archetype    = applicationDefinition.Pipeline.Archetype
+
+        _assertMajorVersionMatchesReleaseVersion(majorVersion)
+
+    }
+}
+
+def _assertMajorVersionMatchesReleaseVersion(majorVersion) {
+
+    relBranchNm = env.BRANCH_NAME.split('/')[-1]
+
+    if (majorVersion.toString() != relBranchNm) {
```

Comment:
could we change the line to : if (!majorVersion.toString().equals(relBranchNm))

Reply:
Maybe. I just realized I should have asked this question about a similar proposed change... what benefit is there to make the change you propose if both approaches work?

Reply:
This is unnecessary. In groovy, `==` is mapped to `.equals()`.

Reply:
`==` is much easier on the :eyes: in my opinion, so I agree with @skinitimski